prolog.cl

```diff
@@ -841,6 +842,747 @@
+           )
+)
+  
+  (multi "A Data Engineer is writing a Python script using the Snowflake Connector for Python. The Engineer will use the snowflake.connector.connect function to connect to Snowflake.
+The requirements are:
+1. Raise an exception if the specified database, schema, or warehouse does not exist
+2. Improve download performance
+Which parameters of the connect function should be used? (Choose two.)
+"
+         5
+        '("client_prefetch_threads"
+          "validate_default_parameters")
+        
+         '("authenticator"
+           "arrow_number_to_decimal"
+           "client_session_keep_alive"
+           )
+)
+
+
+  (multi "How can a relational data be transformed into semi-structured data using the LEAST amount of operational overhead?"
+         4
+        '("Use the OBJECT_CONSTRUCT function to return a Snowflake object.")
+        
+         '("Use the PARSE_JSON function to produce a VARIANT value."
+           "Use the TO_JSON function."
+           "Use the TO_VARIANT function to convert each of the relational columns to VARIANT."
+           )
+)
+
+  (multi "A Data Engineer ran a stored procedure containing various transactions. During the execution, the session abruptly disconnected, preventing one transaction from committing or rolling back. The transaction was left in a detached state and created a lock on resources.
+What step must the Engineer take to immediately run a new transaction?"
+         4
+        '("Call the system function SYSTEM$ABORT_TRANSACTION.")
+        
+         '("Call the system function SYSTEM$CANCEL_TRANSACTION."
+           "Set the LOCK_TIMEOUT to FALSE in the stored procedure."
+           "Set the TRANSACTION_ABORT_ON_ERROR to TRUE in the stored procedure."
+           )
+)
+
+  (multi "Which methods will trigger an action that will evaluate a DataFrame? (Choose two.)"
+         5
+        '("DataFrame.collect()"
+          "DataFrame.show()")
+        
+         '("DataFrame.random_split()"
+           "DataFrame.select()"
+           "DataFrame.col()"
+           )
+)
+
+  (multi "Given the table SALES which has a clustering key of column CLOSED_DATE, which table function will return the average clustering depth for the SALES_REPRESENTATIVE column for the North American region?"
+         4
+        '("select system$clustering_depth('Sales', 'sales_representative', 'region = ''North America''');")
+        
+         '("select system$clustering_information('Sales', 'sales_representative', 'region = ''North America''');"
+           "select system$clustering_depth('Sales', 'sales_representative') where region = 'North America';"
+           "select system$clustering_information('Sales', 'sales_representative') where region = 'North America’;"
+           )
+)
+
+  (multi "A new CUSTOMER table is created by a data pipeline in a Snowflake schema where MANAGED ACCESS is enabled.
+Which roles can grant access to the CUSTOMER table? (Choose three.)"
```

Comment:
Remove the "Choose three" instruction.  I'd also consider dropping the choice selection from 6 to 3-5